pulp_rpm/tests/functional/api/test_acs.py

```diff
@@ -0,0 +1,100 @@
+from pulp_smash import config
+
+from pulp_smash.pulp3.bindings import (
+    delete_orphans,
+    monitor_task,
+    monitor_task_group,
+    PulpTestCase,
+)
+from pulp_smash.pulp3.utils import (
+    gen_repo,
+    get_content_summary,
+)
+
+from pulp_rpm.tests.functional.constants import (
+    PULP_FIXTURES_BASE_URL,
+    RPM_FIXTURE_SUMMARY,
+    RPM_KICKSTART_FIXTURE_SUMMARY,
+    RPM_KICKSTART_ONLY_META_FIXTURE_URL,
+    RPM_ONLY_METADATA_REPO_URL,
+)
+from pulp_rpm.tests.functional.utils import gen_rpm_client, gen_rpm_remote
+
+from pulpcore.client.pulp_rpm import (
+    AcsRpmApi,
+    RemotesRpmApi,
+    RepositoriesRpmApi,
+    RpmRepositorySyncURL,
+)
+
+
+class AlternateContentSourceSyncTestCase(PulpTestCase):
+    """
+    Test Alternate Content Source.
+
+    1. Create ACS
+    2. Sync from repo where only metadata is
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = gen_rpm_client()
+        cls.acs_api = AcsRpmApi(cls.client)
+        cls.repo_api = RepositoriesRpmApi(cls.client)
+        cls.remote_api = RemotesRpmApi(cls.client)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Tear down."""
+        delete_orphans()
+
+    def do_test(self, acs_url, paths, remote_url):
+        """Sync with ACS test."""
+        # ACS is rpm-unsigned repository which has all packages needed
+        acs_remote = self.remote_api.create(gen_rpm_remote(url=acs_url, policy="on_demand"))
+        self.addCleanup(self.remote_api.delete, acs_remote.pulp_href)
+
+        acs_data = {
+            "name": "alternatecontentsource",
+            "remote": acs_remote.pulp_href,
+            "paths": paths,
+        }
+        acs = self.acs_api.create(acs_data)
+        self.addCleanup(self.acs_api.delete, acs.pulp_href)
+
+        acs_refresh = self.acs_api.refresh(acs.pulp_href, acs)
+        monitor_task_group(acs_refresh.task_group)
+
+        # Sync repository with metadata only
+        repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+        remote = self.remote_api.create(gen_rpm_remote(url=remote_url))
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
```

Comment:
I suggest to trigger repo sync before acs refresh and assert it fails, then assert sync is successful after acs refresh.